### PR TITLE
Add a body assignment method for rxUi objects

### PR DESCRIPTION
**Pull request by billdenney**
This replaces nlmixr2/nlmixr2lib#33 with a slightly more general implementation that can be shared across many parts that may be able to use it.

For example, by putting it here, we could use it in `nlmixr2lib` for its function modification and `nlmixr2extra` to simplify some of the covariate search code.

### CODE REVIEW

File: R/function-body.R [hunk 1]
```diff
@@ -0,0 +1,109 @@
+setOldClass("rxUi")
+
+bodySetRxUi <- function(fun, envir = environment(fun), value) {
+  if (is.function(value)) {
+    value <- body(value)
+  }
+  dropEnv <-
+    c(
```
Comment (mattfidler) [inline R/function-body.R:8]: These are hard coded.... I don't think they should be, it should be a white list not a black list

Comment (mattfidler) [reply R/function-body.R:8]: Or should be allowed to be removed by some heuristic.

**Review by mattfidler — COMMENTED** [review]
This simply "modifies" the body and allows things to be retained inside of it?

File: R/function-body.R (hunk 1 above)
Comment (billdenney) [reply R/function-body.R:8]: @mattfidler, I don't know what the correct list should be.  I copied this from nlmixr2/nlmixr2lib#33.  If you have a better way to do it, can you please let me know what that is specifically (or make a commit to this PR to do it)?

**Comment by billdenney** [comment]
> This simply "modifies" the body and allows things to be retained inside of it?

Yes.  The simple way to think of it is that you can change the model itself (`ini()` and `model()`) while retaining everything around it like data, estimation methods, etc.

From my perspective, this is how I'd like to handle any function modification methods for nlmixr2/rxode2:

1. pull the function out with `as.function()`
2. modify the extracted function by traversing the parse tree and making any desired modifications there
3. replace the function body back with `body(ui) <- newFunction`

**Comment by mattfidler** [comment]
Its failing the environment is defunct on my version of ubuntu

**Comment by billdenney** [comment]
I was trying to isolate that.  I didn't think of interactive vs batch mode.

**Comment by mattfidler** [comment]
I think it is because `environment(fun)` can return `NULL`

**Comment by mattfidler** [comment]
However according to docs:

```
     If ‘fun’ is a function or a formula then ‘environment(fun)’
     returns the environment associated with that function or formula.
     If ‘fun’ is ‘NULL’ then the current evaluation environment is
     returned.
```

So it shouldn't...

**Comment by mattfidler** [comment]
Failed on my system.

**Comment by mattfidler** [comment]
This will not work in the long run; 

see:

https://github.com/wch/r-source/blob/e939a5b8c0c942e23efb9c26a1f7229d929504c8/src/library/base/R/formals.R#L33-L42

It will be taken away.  There is also no way to suppress the warnings that are happening.

**Comment by mattfidler** [comment]
The problem is in the `body<-` in the R code I believe

**Comment by billdenney** [comment]
Thank you for taking this forward.  I was at a bit of a loss, and it seems like your impression was similar to mine about how this was turning to be near-impossible using `body<-`.

**Comment by mattfidler** [comment]
This is ready Bill; now I need to rewrite most of `nonmem2rx` handling.

**Comment by mattfidler** [comment]
Nevermind: not ready; looking at `nonmem2rx` I also need rename handling.

**Comment by mattfidler** [comment]
It is mostly done now, I think.

The API is important and this may be a bit easier to understand for many people, so it may be an overall improvement for most.  Though the timelines make it tight.

Before it is done, I guess I should check for proper input and allow strings to be input as well as quoted expressions.

Also maybe `rxUiCompress()`/`rxUiDecompress()` should be more careful of the object class.

The `$sticky`  needs to be documented somewhere, as with everything else... But for now it will be a hidden `API` item.

**Comment by mattfidler** [comment]
Need to merge to test the other packages.

**Pull request merged**